src/doc/man/cargo-package.md

```diff
@@ -81,8 +81,45 @@ There is no guarantee that the source code in the tarball matches the VCS inform
 
 {{#options}}
 
-{{#option "`-l`" "`--list`" }}
+{{#option "`-l`" "`--list`_[fmt]_" }}
 Print files included in a package without making one.
+
+Specifying an output format (rather than the default) is unstable and requires
+`-Zunstable-options`. Valid output formats:
+
+- `json` (unstable, requires `-Zunstable-options`): Emit machine-readable JSON
+  information about each file.
+  ```javascript
+  {
+    /* The Package ID Spec of the package. */
+    "path+file:///home/foo#0.0.0": {
```

Comment:
Like I said at https://github.com/rust-lang/cargo/pull/15311/files#r1995637878, should we do a message per package or a message per file?

Reply:
I am good with per package.

How would message per file look like? Tools may need more work to associate all of them, but tools usually want information for the entire package I assume?
Also, with message-per-file, Cargo then needs to guarantee that there is no interleaving in between. Otherwise tools either need to consume all messages to determine if everything in a single package is collected.